Src/FluentAssertions/Equivalency/DataSetEquivalencyStep.cs

```diff
@@ -161,52 +161,63 @@ private static void CompareTables(IEquivalencyValidationContext context, IEquiva
         {
             if (selectedMembers.ContainsKey(nameof(expectation.Tables)))
             {
-                AssertionScope.Current
+                bool success = AssertionScope.Current
                     .ForCondition(subject.Tables.Count == expectation.Tables.Count)
                     .FailWith("Expected {context:DataSet} to contain {0}, but found {1} table(s)", expectation.Tables.Count, subject.Tables.Count);
 
-                if (dataConfig is not null)
+                if (success)
```

Comment:
🔧 Maybe jumping out of the method when `success` is `false` may make this method better readable.

Reply:
As in an early return?
I would be happy to do that instead, I only avoided it for your sake.

Reply:
In this case, it would make total sense.